deployment/terraform/static.tf

```diff
@@ -0,0 +1,166 @@
+# Static website
+
+resource "aws_s3_bucket" "marlowe_playground" {
+  bucket = "marlowe-playground-website"
+  acl    = "public-read"
+
+  website {
+    index_document = "index.html"
+    error_document = "error.html"
+  }
+}
+
+resource "aws_s3_bucket_policy" "marlowe_playground_website" {
+  bucket = aws_s3_bucket.marlowe_playground.id
+
+  policy = <<POLICY
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Sid": "PublicReadGetObject",
+            "Effect": "Allow",
+            "Principal": "*",
+            "Action": [
+                "s3:GetObject"
+            ],
+            "Resource": [
+                "arn:aws:s3:::marlowe-playground-website/*"
+            ]
+        }
+    ]
+}
+POLICY
+}
+
+resource "aws_iam_role" "s3_proxy_role" {
+  name               = "marlowe_website_s3_${var.env}"
+  path               = "/"  
+  assume_role_policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Action": "sts:AssumeRole",
+      "Principal": {
+        "Service": "apigateway.amazonaws.com"
+      },
+      "Effect": "Allow",
+      "Sid": ""
+    }
+  ]
+}
+EOF
+}
+
+# Root path
+resource "aws_api_gateway_method" "marlowe_root_get_method" {
+  rest_api_id      = aws_api_gateway_rest_api.marlowe_symbolic_lambda.id
+  resource_id      = aws_api_gateway_rest_api.marlowe_symbolic_lambda.root_resource_id
+  http_method      = "GET"
+  authorization    = "NONE"
+  api_key_required = false
+}
+
+resource "aws_api_gateway_integration" "marlowe_root_get_method" {
+  rest_api_id      = aws_api_gateway_rest_api.marlowe_symbolic_lambda.id
+  resource_id      = aws_api_gateway_rest_api.marlowe_symbolic_lambda.root_resource_id
+  http_method      = aws_api_gateway_method.marlowe_root_get_method.http_method
+
+  type                    = "AWS"
+  integration_http_method = "GET"
+  credentials             = "${aws_iam_role.s3_proxy_role.arn}"
+  uri                     = "arn:aws:apigateway:${var.aws_region}:s3:path/marlowe-playground-website/${var.env}/index.html"
+}
+
+resource "aws_api_gateway_method_response" "marlowe_root_get_method" {
+  rest_api_id = aws_api_gateway_rest_api.marlowe_symbolic_lambda.id
+  resource_id      = aws_api_gateway_rest_api.marlowe_symbolic_lambda.root_resource_id
+  http_method = aws_api_gateway_method.marlowe_root_get_method.http_method
+  status_code = "200"
+
+  response_parameters = {
+    "method.response.header.Access-Control-Allow-Headers" = true
+    "method.response.header.Access-Control-Allow-Methods" = true
+    "method.response.header.Access-Control-Allow-Origin"  = true
+    "method.response.header.Content-Type" = true
+  }
```

Comment:
You know, tightening up the allow-origin headers on this might be the easiest first step in improving our security footprint.

Reply:
yeah, I'm not sure what to do at the moment, I think that in the near future this can go away and we can have very strict CORS but that needs to wait until I've proxied the servers behind API gateway

Reply:
That's fair. :-)